src/public/modules/forms/admin/controllers/collaborator-modal.client.controller.js

```diff
@@ -88,20 +91,25 @@ function CollaboratorModalController(
   }
 
   /**
-   * Calls FormAPI update to update the permission list of a form
+   * Calls AdminFormService to update the permission list (collaborators) of a form
    * @param {Array} permissionList - New permission list for the form
    */
   $scope.updatePermissionList = (permissionList) => {
-    return FormApi.update(
-      { formId: $scope.myform._id },
-      { form: { permissionList } },
-    )
-      .$promise.then((savedForm) => {
-        $scope.myform = savedForm
+    return $q
+      .when(
+        AdminFormService.updateCollaborators($scope.myform._id, {
+          permissionList,
+        }),
+      )
+      .then((updatedCollaborators) => {
+        $scope.myform = {
+          ...$scope.myform,
+          ...updatedCollaborators,
+        }
         externalScope.refreshFormDataFromCollab($scope.myform)
       })
       .catch((err) => {
-        Toastr.error(err.data.message)
+        Toastr.error(err.message)
```

Comment:
have you tested this? may need to be `err.response.data.message`, or even better, `_.get(err, 'response.data.message, 'Some default message')` since we don't trust types on the frontend

Reply:
great catch! will read up on axios and fix + document this!